src/main/java/org/fenixedu/academic/service/filter/enrollment/ClassEnrollmentAuthorizationFilter.java

```diff
@@ -108,19 +146,21 @@ private FenixServiceException verify(StudentCurricularPlan studentCurricularPlan
         return result;
     }
 
-    private FenixServiceException verify(DegreeCurricularPlan degreeCurricularPlan) {
-        final EnrolmentPeriodInClasses enrolmentPeriodInClasses = degreeCurricularPlan.getCurrentClassesEnrollmentPeriod();
-        if (enrolmentPeriodInClasses == null || enrolmentPeriodInClasses.getStartDateDateTime() == null
-                || enrolmentPeriodInClasses.getEndDateDateTime() == null) {
+    private FenixServiceException verify(DegreeCurricularPlan degreeCurricularPlan, ExecutionSemester executionSemester) {
+        final Optional<EnrolmentPeriod> enrolmentPeriodInClasses =
+                degreeCurricularPlan.getClassesEnrollmentPeriod(executionSemester);
+        if (!enrolmentPeriodInClasses.isPresent() || enrolmentPeriodInClasses.get().getStartDateDateTime() == null
```

Comment:
It is possible to use some of that but in this case there are 3 different possible outcomes and it would be somewhat confusing doing it through that way in the entire method